### PR TITLE
fix: Hot reloading with chokidar & busting of the require cache.

### PR DESCRIPTION
**Pull request by robblovell**
### Hot Reload

This is a fix for issue #505, hot reload not working.

The solution applied here is to use chokidar to watch files in the distribution/bundle folder, wether that be esbuild, webpack, or typescript doing the build. 

Upon a rebuild, the node.js require cache is busted for the changed handler 'extractBehaviors()' is re-called to reload the handlers. The reload is via an import statement which caches imported code, so a cache bust is required for the newly compiled/bundled code to be loaded.

Chokidar is debounced so that the handlers are reloaded at the end of a build. Any file in the directory specified by the "path" option will be watched and if changed hot-reloaded. Note that the default path is the current directory (".").

I use esbuild and have not tested this with typescript based builds, but the same principles will apply and it should work.

The watch/reload mechanism is disabled by default so that current usage is unchanged. The flag "watchReload: true" will turn on the watcher so that typescript and esbuild solutions use the watcher to hot reload the handlers.

example:
```
   offlineEdgeLambda: {
      path: '.esbuild/.build',
      watchReload: true,
    },
```

Additional options can be used to modify the behavior of the file watcher and debounce logic (ignoreInitial, awaitWriteFinish, interval, debounce). 

For example:
```
   offlineEdgeLambda: {
      path: '.esbuild/.build',
      watchReload: true,
      ignoreInitial: true,
      awaitWriteFinish: true,
      interval: 500,
      debounce: 750,
   },
```

Implementations using webpack appears to have a solution already in place for hot reload. However, my experimentation with the server restart mechanism used by the webpack solution's onReload() function appears to not work very well. It might be better for the onReload() to call extractBehaviors() instead of server.restart(). This could be implemented in a future pull request.

### CODE REVIEW

**Comment by robblovell** [comment]
Somehow I managed to close this off unintentionally. I am reopening.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "serverless-offline-edge-lambda",
-  "version": "1.2.0",
```
Comment (robblovell) [inline package.json:3]: Is this manually incremented or should this be put back to 1.2.0 for an auto increment?

File: src/behavior-router.ts [hunk 2]
```diff
@@ -55,13 +56,32 @@ export class BehaviorRouter {
 		this.cfResources = serverless.service?.resources?.Resources || {};
 		this.cacheDir = path.resolve(options.cacheDir || path.join(os.tmpdir(), 'edge-lambda'));
 		this.fileDir = path.resolve(options.fileDir || path.join(os.tmpdir(), 'edge-lambda'));
-		this.path = this.serverless.service.custom.offlineEdgeLambda.path || '';
+		this.path = this.serverless.service.custom.offlineEdgeLambda.path || '.';
```
Comment (robblovell) [inline src/behavior-router.ts:59]: I believe changing the default from '' to '.' should have no effect. However, it has not been explicitly tested.

**Comment by robblovell** [comment]
The readme should be updated to document the new options for hot reload.

File: src/behavior-router.ts (hunk 2 above)
Comment (robblovell) [reply src/behavior-router.ts]: Opted to not change the default and use path.join instead.

**Review by mattstrom — APPROVED** [review]
Thanks for the contribution! I resolved two small conflicts in _package.json_ and _package-lock.json_

**Pull request merged**